peer/gossip/mcs.go

```diff
@@ -185,6 +187,28 @@ func (s *MSPMessageCryptoService) VerifyBlock(chainID common.ChainID, seqNum uin
 	return policy.Evaluate(signatureSet)
 }
 
+func (s *MSPMessageCryptoService) VerifyHeader(chainID string, signedBlock *pcommon.Block) error {
+	if signedBlock == nil {
+		return fmt.Errorf("Invalid Block on channel [%s]. Block is nil.", chainID)
+	}
+
+	if signedBlock.Header == nil {
+		return fmt.Errorf("Invalid Block on channel [%s]. Header must be different from nil.", chainID)
+	}
+
+	// - Unmarshal metadata
+	if signedBlock.Metadata == nil || len(signedBlock.Metadata.Metadata) == 0 {
+		return fmt.Errorf("Block with id [%d] on channel [%s] does not have metadata. Block not valid.", signedBlock.Header.Number, chainID)
+	}
+
+	metadata, err := utils.GetMetadataFromBlock(signedBlock, pcommon.BlockMetadataIndex_SIGNATURES)
```

Comment:
we need to check bounds because GetMetadataFromBlock doesn't do it for you.

Reply:
the signatures are at index 0 so if the len is >0, it will not be out of bounds. Just like in VerifyBlock(), above.